docs/ruby_tutorial.md

```diff
@@ -0,0 +1,771 @@
+<!-- STOP
+  This file was generated by a rundoc script, do not modify it.
+
+  Instead modify the rundoc script and re-run it.
+
+  Command: bin/rundoc build --path test/fixtures/cnb/ruby/rundoc.md
+STOP -->
+
+# Heroku Ruby Cloud Native Buildpack (CNB) Tutorial
+
+Build a Ruby on Rails application image in 5 minutes, no Dockerfile required.
+
+At the end of this tutorial, you'll have a working [OCI image](https://opencontainers.org/) of a Ruby on Rails application that can run locally. You will learn about the Cloud Native Buildpack (CNB) ecosystem, and how to utilize the [pack CLI](https://buildpacks.io/docs/for-platform-operators/how-to/integrate-ci/pack/) to build images without the need to write or maintain a Dockerfile.
+
+## Install the pack CLI
+
+We assume you have [docker installed](https://docs.docker.com/engine/install/) and a working copy [of git](https://github.com/git-guides/install-git). Next, you will need to install the CLI tool for building CNBs, [pack CLI](https://buildpacks.io/docs/for-platform-operators/how-to/integrate-ci/pack/). If you're on a Mac you can install it via Homebrew:
+
+```
+$ brew install buildpacks/tap/pack
+```
+<!-- STOP. This document is autogenerated. Do not manually modify. See the top of the doc for more details. -->
```

Comment:
What do you think about removing all of these "don't edit this document" comments in favour of having a CI check that performs the auto-generation, and then fails if `git status` reports there are changes in the working directory? (Similar to what the base-images repo does for checking the `installed-packages*.txt` files are up to date.)

Reply:
I took a look at that repo and found https://github.com/heroku/base-images/blob/57678b0bf1f5eaeb4ba1a27392546be171bce39d/.github/workflows/ci.yml#L47-L50. One big difference between docs and that is that the doc output is often non-deterministic. Even if our output is deterministic, other tools like bundler might output text in a different order (since they use threads to download multiple gems at the same time).

We could perhaps do something like the changelog check where it fails any build that modifies one of these docs unless they've set a specific flag on the PR.

If we wanted to skip adding the comments they would need to be manually removed or a feature/configuration added to rundoc to skip that behavior.